### PR TITLE
refactor: no dupe bridge without go template

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_cmd.sh [hunk 1]
```diff
@@ -82,4 +82,5 @@ NEEDS_DOCKER_LOGIN="{{and IsDockerContainerRuntime HasPrivateAzureRegistryServer
 IPV6_DUAL_STACK_ENABLED="{{IsIPv6DualStackFeatureEnabled}}"
 OUTBOUND_COMMAND="{{GetOutboundCommand}}"
 ENABLE_UNATTENDED_UPGRADES="{{EnableUnattendedUpgrade}}"
+ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE="{{ and NeedsContainerd IsKubenet (not HasCalicoNetworkPolicy) }}"
```
Comment (ganeshkumar5699) [inline parts/linux/cloud-init/artifacts/cse_cmd.sh:85]: typo here?

Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:85]: I don't think so:
```
and
	Returns the boolean AND of its arguments by returning the
	first empty argument or the last argument. That is,
	"and x y" behaves as "if x then y else x."
	Evaluation proceeds through the arguments left to right
	and returns when the result is determined.
```

Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:85]: from https://pkg.go.dev/text/template

this confused me at first too

Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:85]: Oh, interesting. Thanks!

**Review by ganeshkumar5699 — APPROVED** [review]

File: pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet+FIPSEnabled/CSECommand [hunk 2]
```diff
@@ -1 +1 @@
...
...BUCKET_WRITE=0 LOAD_BALANCER_DISABLE_OUTBOUND_SNAT=<nil> USE_MANAGED_IDENTITY_EXTENSION=false USE_INSTANCE_METADATA=false LOAD_BALANCER_SKU= EXCLUDE_MASTER_FROM_STANDARD_LB=true MAXIMUM_LOADBALANCER_RULE_COUNT=0 CONTAINER_RUNTIME=containerd CLI_TOOL=ctr CONTAINERD_DOWNLOAD_URL_BASE=https://storage.googleapis.com/cri-containerd-release/ NETWORK_MODE= KUBE_BINARY_URL= USER_ASSIGNED_IDENTITY_ID=userAssignedID API_SERVER_NAME= IS_VHD=true GPU_NODE=false SGX_NODE=false MIG_NODE=false CONFIG_GPU_DRIVER_IF_NEEDED=true ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED=false TELEPORTD_PLUGIN_DOWNLOAD_URL= CONTAINERD_VERSION= CONTAINERD_PACKAGE_URL= RUNC_VERSION= RUNC_PACKAGE_URL= ENABLE_HOSTS_CONFIG_AGENT="false" DISABLE_SSH="false" NEEDS_CONTAINERD="true" TELEPORT_ENABLED="false" SHOULD_CONFIGURE_HTTP_PROXY_CA="false" SHOULD_CONFIGURE_CUSTOM_CA_TRUST="false" IS_KRUSTLET="false" GPU_NEEDS_FABRIC_MANAGER="false" NEEDS_DOCKER_LOGIN="false" IPV6_DUAL_STACK_ENABLED="false" OUTBOUND_COMMAND="curl -v --insecure --proxy-insecure https://mcr.microsoft.com/v2/" ENABLE_UNATTENDED_UPGRADES="true" ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE="true" /usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision_start.sh"
```
Comment (alexeldeib) [inline pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet+FIPSEnabled/CSECommand:1]: @ganeshkumar5699 ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE="true"

File: pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet+Calico/CSECommand [hunk 3]
```diff
@@ -1 +1 @@
...
...UCKET_WRITE=0 LOAD_BALANCER_DISABLE_OUTBOUND_SNAT=<nil> USE_MANAGED_IDENTITY_EXTENSION=false USE_INSTANCE_METADATA=false LOAD_BALANCER_SKU= EXCLUDE_MASTER_FROM_STANDARD_LB=true MAXIMUM_LOADBALANCER_RULE_COUNT=0 CONTAINER_RUNTIME=containerd CLI_TOOL=ctr CONTAINERD_DOWNLOAD_URL_BASE=https://storage.googleapis.com/cri-containerd-release/ NETWORK_MODE= KUBE_BINARY_URL= USER_ASSIGNED_IDENTITY_ID=userAssignedID API_SERVER_NAME= IS_VHD=true GPU_NODE=false SGX_NODE=false MIG_NODE=false CONFIG_GPU_DRIVER_IF_NEEDED=true ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED=false TELEPORTD_PLUGIN_DOWNLOAD_URL= CONTAINERD_VERSION= CONTAINERD_PACKAGE_URL= RUNC_VERSION= RUNC_PACKAGE_URL= ENABLE_HOSTS_CONFIG_AGENT="false" DISABLE_SSH="false" NEEDS_CONTAINERD="true" TELEPORT_ENABLED="false" SHOULD_CONFIGURE_HTTP_PROXY_CA="false" SHOULD_CONFIGURE_CUSTOM_CA_TRUST="false" IS_KRUSTLET="false" GPU_NEEDS_FABRIC_MANAGER="false" NEEDS_DOCKER_LOGIN="false" IPV6_DUAL_STACK_ENABLED="false" OUTBOUND_COMMAND="curl -v --insecure --proxy-insecure https://mcr.microsoft.com/v2/" ENABLE_UNATTENDED_UPGRADES="true" ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE="false" /usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision_start.sh"
```
Comment (alexeldeib) [inline pkg/agent/testdata/AKSUbuntu1804+Containerd+Kubenet+Calico/CSECommand:1]: @ganeshkumar5699 ENSURE_NO_DUPE_PROMISCUOUS_BRIDGE="false" when containerd + kubenet true, (not true -> false for calico)

**Pull request merged**